automation_tools/__init__.py

```diff
@@ -3173,15 +3173,68 @@ def attach_subscription(url=None, consumer=None, pool_id=None, count=None):
         raise ValueError("Pass values under pool_id and count")
 
 
-def relink_manifest(manifest_file=None):
+def refresh_manifest(url, consumer, user, password, exp_subs_file):
+    """Task to remove all subscriptions from manifest and attach required ones.
+
+    :param url: Subscription Manager URL
+    :param consumer: A consumer hash to be used for getting the manifest
+    :param user: Red Hat Network username
+    :param password: Red Hat Network password
+    :param exp_subs_file: Expected subscription file
+    :returns: boolean True/False
+    """
+    auth_details = u'{0}:{1}'.format(user, password)
+
+    if isinstance(auth_details, str):  # py3
+        bytestring = bytes('{0}:{1}'.format(user, password), 'utf-8')
+    else:  # py2
+        bytestring = bytes('{0}:{1}'.format(user, password))
+
+    base64string = base64.encodestring(bytestring).strip()
+
+    command = ('curl -sk -X DELETE -H "Authorization:Basic {0}"'
+               ' "{1}/subscription/consumers/{2}/entitlements"').format(
+        base64string.decode('utf-8'), url, consumer)
+
+    print("Cleaning up all subscriptions from Manifest!")
+    response = run(command)
+    result = 'deletedRecords' in response
+
+    if result:
+        with open(exp_subs_file, 'r') as fp:
+            exp_subs = {}
+            exp_subs_details = [r.strip().split(';') for r in fp.readlines() if not r.strip(
+            ).startswith('#')]
+            for name_list in exp_subs_details:
+                exp_subs[name_list[0]] = name_list[1:]
+
+            for sub in exp_subs.keys():
+                print("Attaching subscription {}".format(sub))
+                attach = attach_subscription(pool_id=exp_subs[sub][0], count=exp_subs[sub][1])
+                if attach:
+                    print("Successfully attached subscription {}".format(sub))
+                else:
+                    return attach
+    return result
+
+
+def relink_manifest(url, consumer, user, password, exp_subs_file, manifest_file=None):
     """Links the latest downloaded manifest file to the manifest_latest.zip
     softlink.
 
+    :param url: Subscription Manager URL
+    :param consumer: A consumer hash to be used for getting the manifest
+    :param user: Red Hat Network username
+    :param password: Red Hat Network password
+    :param exp_subs_file: Expected subscription file
     :param manifest_file: Specify the manifest file path.
     """
     if manifest_file is None:
         manifest_file = download_manifest()
         if os.environ.get('EXP_SUBS_FILE', None) is not None:
```

Comment:
We are still checking the environ variable here

Reply:
@san7ket I have only replaced the the env vars which are part of new change ... however the change which you mention requires a different PR to replace every env vars from  `automation_tools/__init__.py`

Reply:
See my PR https://github.com/SatelliteQE/robottelo-ci/pull/1633
This is now passed as a parameter , So the function would not get the env variable. 
Just check the parameter here, 
`if exp_subs_file is not None:`

Reply:
ahh .. Thanks ..
its there now .